react.md

```diff
@@ -303,64 +303,114 @@ Contains Jest configuration. If you're adding Jest plugins or configuring Enzyme
-        ),
-      phone_number: Yup.string()
-        .matches(
-          /^(\+?\d{0,4})?\s?-?\s?(\(?\d{3}\)?)\s?-?\s?(\(?\d{3}\)?)\s?-?\s?(\(?\d{4}\)?)?$/,
-          'Please enter a valid phone number'
-        )
-        .required('Please enter a valid phone number'),
-    ```
-
-#### Yup: Standards
-- TODO: How do we build Yup here?
+Yup.js is an object schema validator that integrates seamlessly with Formik. With Yup, you build a schema describing the expected shape of your form's data and Formik will take care of running validation on change and propagating errors to your fields. Yup provides a plethora of built-in validators and allows the addition of custom validations using `yup.addMethod`.
+
+Here's a quick example of a custom validator and a signup form schema.
+
+#### utils.yup.js
+```js
+// in a utils/yup.js file
+import * as yup from 'yup';
+
+// Set a custom default required message
+yup.setLocale({
+  mixed: {
+    required: ({ path: label }) => `Please enter a valid ${label}`,
+  }
+})
+
+yup.addMethod(yup.string, 'password', function validatePassword(length, message) {
+  return this.test('password', message, function testPassword(value) {
+    const { path, createError } = this;
+    const valid = value.match(yourPasswordRegexHere);
+
+    return new Promise((resolve, reject) => {
+      if (valid) {
+        resolve(true);
+      } else {
+        reject(createError({
+          path,
+          message: message || `${path} must have ${length} at least one number, one letter, and one symbol`,
+        }));
+      }
+    });
+  });
+});
+
+export default yup;
+```
+
+#### FormComponent.js
+```js
+import { PHONE_NUMBER } from 'utils/regex';
+import yup from 'utils/yup';
+
+const SignupSchema = yup.object({
+  first_name: yup.string()
+    .label('first name')
+    .min(2)
+    .required(),
+  email: yup.string()
+    .email()
+    .required(),
+  password: yup.string()
+    .min(8)
+    .max(128)
+    .password(),
+  password_confirm: yup.string()
+    .label('password')
+    .required(({ path: label }) => `Those two ${label}s do not match. Please try again.`)
+    .oneOf(
+      [yup.ref('password'), null],
+      ({ path: label }) => `Those two ${label}s do not match. Please try again.`
+    ),
+  phone_number: yup.string()
+    .label('phone number')
+    .matches(
+      PHONE_NUMBER,
+    )
+    .required(),
+});
+```
+
+#### Yup: Best Practices
+- Make liberal use of `yup.addMethod` & descriptive validator names to avoid repeating validation code. ie. `birthDate.string().isAdult()` is always easier to understand at a glance than a series of min/max validations full of date parsing code. Additionally, custom validators are easily unit testable, while one-off custom validation rules in a schema are not.
+- Any regexes used with `matches` should be named constants, even if you think it's obvious what the regex does.
+- Make use of `label()` to give your fields human-friendly names in error messages, if necessary.
```

Comment:
I didn't know about the `label()` or `...setLocale` properties. Thank you for the explanation as well.